var/spack/repos/builtin/packages/py-maestrowf/package.py

```diff
@@ -21,6 +25,6 @@ class PyMaestrowf(PythonPackage):
     depends_on('py-setuptools', type='build')
     depends_on('py-pyyaml',     type=('build', 'run'))
     depends_on('py-six',        type=('build', 'run'))
-    depends_on('py-enum34',     type=('build', 'run'))
+    depends_on('py-enum34',     type=('build', 'run'), when='^python@:3.3')
```

Comment:
I noticed that older versions of maestrowf do require enum34 regardless of Python version. If you want, you can do something like:
```python
depends_on('py-enum34', type=('build', 'run'), when='^python@:3.3')
depends_on('py-enum34', type=('build', 'run'), when='@:1.1.3')
```
If either of these conditions are met, the dependency will be added. Otherwise, old versions will probably cause setuptools to download the dependency themselves.

Reply:
> I noticed that older versions of maestrowf do require enum34 regardless of Python version. If you want, you can do something like:
> 
> ```python
> depends_on('py-enum34', type=('build', 'run'), when='^python@:3.3')
> depends_on('py-enum34', type=('build', 'run'), when='@:1.1.3')
> ```
> 
> If either of these conditions are met, the dependency will be added. Otherwise, old versions will probably cause setuptools to download the dependency themselves.

@adamjstewart -- That's a fine point. I was actually wondering about that since the older versions do accidentally require it. I'll update that.